src/zoomifytilesource.js

```diff
@@ -0,0 +1,97 @@
+(function( $ ){
+
+  /**
+   * @class ZoomifyTileSource
+   * @classdesc A tilesource implementation for the zoomify format.
+   *
+   * @memberof OpenSeadragon
+   * @extends OpenSeadragon.TileSource
+   * @param {Number} width - the pixel width of the image.
+   * @param {Number} height
+   * @param {Number} tileSize
+   * @param {String} tilesUrl
+   */
+  $.ZoomifyTileSource = function( options ) {
+
+  	options.tileSize = 256;
+		
+		var currentImageSize = {x:options.width,y:options.height};
+		options.imageSizes = [{x:options.width,y:options.height}];
+  	options.gridSize = [this._getGridSize(options.width,options.height,options.tileSize)];
+		
+  	while(parseInt(currentImageSize.x) > options.tileSize || parseInt(currentImageSize.y) > options.tileSize) {
```

Comment:
While it may not be entirely necessary in this case, it's a good habit to always add the radix to your parseInt calls, e.g. `parseInt(currentImageSize.x, 10)`.